src/Core/src/Signer/SignerV4.php

```diff
@@ -101,6 +96,24 @@ protected function buildBodyDigest(Request $request, bool $isPresign): string
         return $hash;
     }
 
+    protected function convertBodyToStream(Request $request, \DateTimeImmutable $now, string $credentialString, string $signingKey, string &$signature): void
+    {
+        $body = $request->getBody();
+        if ($request->hasHeader('content-length')) {
+            $contentLength = (int) $request->getHeader('content-length');
+        } else {
+            $contentLength = $body->length();
```

Comment:
Why do we need the length here?

Reply:
You're totally right.. This method is now, almost empty..
I've refactored it to remove all those arguments, and remove the byref variable.
Please review

Reply:
I really like the SigningContext. :+1: